.travis.yml

```diff
@@ -1,36 +1,63 @@
 language: node_js
+notifications:
+  email:
+    recipients:
+    - bugs@unchained-capital.com
+    on_success: change
+    on_failure: always
 
 jobs:
   include:
-    - stage: test
+    - stage: "Tests"
+      node_js: 8
+      name: "Node JS 8"
       script: npm run test
-      node_js: 10
+    - node_js: 10
+      name: "Node JS 10"
       script: npm run test
     - node_js: 12
+      name: "Node JS 12"
       script: npm run ci
-
+    - stage: "Cypress Tests"
+      node_js: 10
+      addons:
+        apt:
+          packages:
+            # Ubuntu 16+ does not install this dependency by default, so we need to install it ourselves
+            - libgconf-2-4
+      cache:
+        # Caches $HOME/.npm when npm ci is default script commandK
+        # Caches node_modules in all other cases
+        npm: true
+        directories:
+          # we also need to cache folder with Cypress binary
+          - ~/.cache
+      install:
+        - npm ci
+      before_script:
+        - npm run build:ci
+        # The server from create-react-app doesn't work well with travis
+        - npm i -g http-server
+        # Necessary because create-react-app assumes you want to deploy to /caravan
+        - mkdir ./build/caravan
+        - shopt -s extglob
+        - mv ./build/!(caravan) ./build/caravan/
+        - http-server build -p 3000 -c-1 --silent &
```

Comment:
The above four lines are a little ugly but I couldn't figure out another way around this.